experimental/dds/tree2/src/shared-tree-core/branch.ts

```diff
@@ -446,22 +446,14 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 	 *
 	 * @param branch - the branch to rebase onto
 	 * @param upTo - the furthest commit on `branch` over which to rebase (inclusive). Defaults to the head commit of `branch`.
-	 * @returns the net change to this branch and the commits that were removed and added to this branch by the rebase,
-	 * or undefined if nothing changed
 	 */
-	public rebaseOnto(
-		branch: SharedTreeBranch<TEditor, TChange>,
-		upTo = branch.getHead(),
-	):
-		| [
-				change: TChange | undefined,
-				removedCommits: GraphCommit<TChange>[],
-				newCommits: GraphCommit<TChange>[],
-		  ]
-		| undefined {
+	public rebaseOnto(branch: SharedTreeBranch<TEditor, TChange>, upTo = branch.getHead()): void {
```

Comment:
It's not a huge deal (because as you pointed out, they aren't used), but the return values of each of these functions is supposed to be the same information that you get from the corresponding event firing. That way, if you want to do an operation and immediately examine the results you can, rather than having to subscribe to an event.

Since we don't actually do any of that, it probably makes sense to remove the return value from _all_ of these methods, so that it's consistent. Or, maybe it's easy to leave it as-is if you do end up changing it to a thunk.

Reply:
Yeah, having them around does increase the testability of the code.